lib/types/string.js

```diff
@@ -662,6 +662,13 @@ module.exports = Any.extend({
                         return helpers.error('string.domain', { value: matched });
                     }
 
+                    if(options?.domain?.hostname){
```

Comment:
`domain` is already checked a few lines above, you can probably refactor that if to handle your case.

Reply:
Are you talking about the domain variable few lines above (line 659)?
I think it's not the same as options.domain...

Reply:
Technically, it kind of is. Look at the `$_addRule` just above.

`options` (the 3rd argument of `validate`) is basically the raw argument you provided.

`domain` (the 4th argument) is the post-processed version of that raw argument through the call to `addressOptions`, which I think should be augmented to validate that `hostname` satisfies expectations.

Reply:
I added support for arrays of hostnames. What kind of changes would be useful in order to support regex?